src/main/java/org/openrewrite/java/testing/mockito/MockitoWhenOnStaticToMockStatic.java

```diff
@@ -0,0 +1,92 @@
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.internal.ListUtils;
+import org.openrewrite.java.JavaIsoVisitor;
+import org.openrewrite.java.JavaParser;
+import org.openrewrite.java.JavaTemplate;
+import org.openrewrite.java.MethodMatcher;
+import org.openrewrite.java.search.UsesType;
+import org.openrewrite.java.tree.*;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class MockitoWhenOnStaticToMockStatic extends Recipe {
+
+    private static final MethodMatcher MOCKITO_WHEN = new MethodMatcher("org.mockito.Mockito when(..)");
+
+    @Override
+    public String getDisplayName() {
+        return "";
+    }
+
+    @Override
+    public String getDescription() {
+        return ".";
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getVisitor() {
+        return Preconditions.check(new UsesType<>("org.mockito.Mockito", true),
+                new JavaIsoVisitor<ExecutionContext>() {
+                    @Override
+                    public J.MethodDeclaration visitMethodDeclaration(J.MethodDeclaration method, ExecutionContext ctx) {
+                        J.MethodDeclaration m = super.visitMethodDeclaration(method, ctx);
+                        boolean rewrittenWhen = false;
+                        List<Statement> statementsBeforeWhen = new ArrayList<>();
+                        List<Statement> statementsAfterWhen = new ArrayList<>();
+                        for (Statement stmt : m.getBody().getStatements()) {
+                            if (stmt instanceof J.MethodInvocation &&
+                                MOCKITO_WHEN.matches(((J.MethodInvocation) stmt).getSelect())) {
+                                J.MethodInvocation when = (J.MethodInvocation) ((J.MethodInvocation) stmt).getSelect();
+                                if (when.getArguments().get(0) instanceof J.MethodInvocation && ((J.MethodInvocation) when.getArguments().get(0)).getMethodType().getFlags().contains(Flag.Static)) {
+                                    JavaType.FullyQualified arg_fq = TypeUtils.asFullyQualified(when.getArguments().get(0).getType());
+                                    String template = String.format("try(MockedStatic<%s> mock%s = mockStatic(%s.class)){\n" +
+                                                                    "mock%s.when(%s::%s).thenReturn(%s);\n" +
+                                                                    "}", arg_fq.getClassName(), arg_fq.getClassName(), arg_fq.getClassName(), arg_fq.getClassName(), arg_fq.getClassName(), ((J.MethodInvocation) when.getArguments().get(0)).getSimpleName(), ((J.MethodInvocation) stmt).getArguments().get(0));
+                                    m = JavaTemplate.builder(template)
+                                            .contextSensitive()
+                                            .javaParser(JavaParser.fromJavaVersion())
+                                            .staticImports("org.mockito.Mockito.mockStatic")
```

Comment:
It looks like you're missing an import here, which will have a cascading effect of missing types.
```suggestion
                                            .imports("org.mockito.MockedStatic")
                                            .staticImports("org.mockito.Mockito.mockStatic")
```